### PR TITLE
feat(vercel): update installation flow to handle multi-org and logged out users

### PR DESCRIPTION
**Pull request by scefali**
This PR adds logic to handle two edge cases for the Vercel external installation flow: logged out users and users with multiple orgs. If a user is logged out, we send them to the login page and encode the next URL as the query parameter `next`. Once they log in, they will go back to the page they started.

If they have multiple organizations, we redierct them a UI where they can pick an organization. Once that flow is complete, they get redirected back to the original URL with a new query parameter of `orgSlug` which represents the organization they chose. The Django view for the configure view will complete the installation with the org specified in the query parameter.

Note that we have somewhat similar flows with Github and VSTS. I have written the code in such a way we can adapt it to those uses cases by extending the view classes (one in Django and one in React). At some point in the near future, I will switch Githhub and VSTS to use these new views.

Screenshots of the new React view:

Screen Shot 2020-06-16 at 10 05 15 AM

Screen Shot 2020-06-16 at 10 05 43 AM

### CODE REVIEW

File: src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+type Props = RouteComponentProps<{integrationSlug: string}, {}>;
+
+type State = AsyncView['state'] & {
+  selectedOrg?: string;
+  organization?: Organization;
+  provider?: IntegrationProvider;
+};
+
+//TODO: make generic props so we can use with Github external installation by extending class
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx:31]: 👎 to extending a react component. I would highly prefer composition over inheritance, so if anything I would say the TODO would be to split this into a reusable component.

Comment (scefali) [reply src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx:31]: @EvanPurkhiser good call out, I'll change it

File: src/sentry/web/frontend/vercel_extension_configuration.py [hunk 2]
```diff
@@ -1,32 +1,13 @@
...
-from sentry.web.frontend.base import BaseView
 from sentry import features
+from .integration_extension_configuration import IntegrationExtensionConfigurationView
 
 
-class VercelExtensionConfigurationView(BaseView):
-    auth_required = False
+class VercelExtensionConfigurationView(IntegrationExtensionConfigurationView):
+    provider = "vercel"
+    external_provider_key = "vercel"
```
Comment (MeredithAnya) [inline src/sentry/web/frontend/vercel_extension_configuration.py:10]: when would these be different?

Comment (scefali) [reply src/sentry/web/frontend/vercel_extension_configuration.py:10]: @MeredithAnya for VSTS, the value of `external_provider_key` is `vsts-extension`

File: src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+      {
+        eventKey: 'integrations.integration_viewed',
+        eventName: 'Integrations: Integration Viewed',
+        integration_type: 'first_party',
+        integration: provider.key,
+        //We actually don't know if it's installed but neither does the user in the view and multiple installs is possible
+        already_installed: false,
+        view: 'external_install',
+      },
+      organization,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx:58]: I wouldn't mind pulling out these parameters into a `get baseTrackPrams`

```tsx
get baseTrackParams() {
    const {organization, provider} = this.state;
    //should have these set but need to make TS happy
    if (!organization || !provider) {
      return;
    }

    return {
      eventKey: 'integrations.integration_viewed',
      eventName: 'Integrations: Integration Viewed',
      integration_type: 'first_party',
      integration: provider.key,
      // We actually don't know if it's installed but neither does
      // the user in the view and multiple installs is possible
      already_installed: false,
      view: 'external_install',
    } as const;
}

trackOpened() {
  const {organization, provider} = this.state;
  //should have these set but need to make TS happy
  if (!organization || !provider) {
    return;
  }

  trackIntegrationEvent(this.baseTrackParams, organization);
}
```

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx:58]: Since you use them twice that is

Comment (scefali) [reply src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx:58]: @EvanPurkhiser good suggestion, I'll make this change

**Review by evanpurkhiser — APPROVED** [review]
lg2m

**Pull request merged**